### DISCUSSION TITLE
When running multiple Milvus instances on the same machine, is it necessary to set independent ports for each component?

### DISCUSSION THREAD

**Discussion by charleskan** [milvus-io/milvus#33290] [Q&A and General discussion] [answered]
I am attempting to build multiple Milvus instances using source code on the same Linux machine. I have noticed that the Milvus configuration file milvus.yaml contains port settings for various components such as rootCoord, proxy, dataCoord, etc.

My question is, when setting up ports for multiple Milvus instances, do I only need to set different ports for the proxy component of each instance, or do I need to set independent ports for all components of each instance?

If all instances' components use the same ports, could this lead to connection errors between components? For example, might one component unintentionally connect to the corresponding component of another Milvus instance?

My current understanding is that to ensure correct communication between components of each Milvus instance, the best practice is to assign independent ports for all components of each instance. This way, each component will know exactly which port to connect to, avoiding the possibility of connecting to components of other instances.

Is this understanding correct? If so, should I set independent ports for all components of each Milvus instance?

If you need any additional information or context, please let me know. Thank you very much for your help and advice!

**Comment by yhmo** [comment, marked as answer]
To avoid potential risk, different milvus instances should use different third-parties(etcd/minio/pulsar).
If two milvus instances share the same etcd(the same rootPath), the same minio(the same bucket and same rootPath), they could delete data of each other and run into trouble.

**Comment by xiaofan-luan** [comment]
The recommended way to do that is to use docker compose.

if you want to deploy multi process in one pod, you should deploy as a standalone
